pkg/builder/builder.go

```diff
@@ -642,10 +657,27 @@ func makeCanaryQueryProbe(vdb *vapi.VerticaDB) *corev1.Probe {
 	}
 }
 
+// getHTTPServerVersionEndpointProbe returns an HTTPGet probe if vclusterops
+// is enabled and server version is newer or equal to v23.4
+func getHTTPServerVersionEndpointProbe(vdb *vapi.VerticaDB) *corev1.Probe {
+	if vmeta.UseVClusterOps(vdb.Annotations) {
+		// if the server version is older than v23.4 then
+		// we return nil
+		vdbVer, ok := vdb.MakeVersionInfo()
```

Comment:
I'm not sure if we will ever have the version the first time. The version isn't known until the pod is running and this needs to be setup before the pod is running. Rather than rely on a specific version, lets just tie it too vclusterops. If vclusterops is enabled we will use this new health probe. That is more or less a check for 24.1.0.